integration-tests/ahm/src/mock.rs

```diff
@@ -90,3 +98,92 @@ pub fn enqueue_dmp(msgs: Vec<InboundDownwardMessage>) {
 		);
 	}
 }
+
+// Migrates the pallet out of the Relay Chain and returns the corresponding Payload.
+//
+// Sends DMP messages with pallet migration data from relay chain to asset hub.
+pub fn rc_migrate<RcMigrator: RcPalletMigrationChecks>(
+	mut relay_chain: RemoteExternalities<PolkadotBlock>,
+) -> (Vec<InboundDownwardMessage>, RcMigrator::RcPayload) {
+	// AH parachain ID
+	let para_id = ParaId::from(1000);
+
+	// Simulate relay blocks and grab the DMP messages
+	let (dmp_messages, rc_payload) = relay_chain.execute_with(|| {
+		let mut dmps = Vec::new();
+
+		if let Ok(stage) = std::env::var("START_STAGE") {
+			let stage = MigrationStage::from_str(&stage).expect("Invalid start stage");
+			RcMigrationStage::<Polkadot>::put(stage);
+		}
+
+		let rc_payload = RcMigrator::pre_check();
+
+		// Loop until no more DMPs are added and we had at least 1
+		loop {
+			next_block_rc();
+
+			let new_dmps =
+				runtime_parachains::dmp::DownwardMessageQueues::<Polkadot>::take(para_id);
+			dmps.extend(new_dmps);
+
+			if RcMigrationStage::<Polkadot>::get() ==
+				pallet_rc_migrator::MigrationStage::MigrationDone
+			{
+				log::info!("Migration done");
+				break (dmps, rc_payload);
+			}
+		}
+	});
+	relay_chain.commit_all().unwrap();
+	// TODO: for some reason this prints some small value (2947), but logs on XCM send and
```

Comment:
We batch messages into XCM messages, about 480 for the accounts migration or so, depending on the migration.